partcad/src/partcad/test/test.py

```
@@ -0,0 +1,57 @@
+    def _log_message_prepare(self, *args) -> str:
+        if args:
+            message = args[0] % args[1:] if args[1:] else args[0]
+            message = f": {message}"
+        else:
+            message = ""
+        return message
+
+    def debug(self, shape, *args) -> None:
+        """This methods works like logging.debug() but prepends the message with the test name and the shape name."""
+        message = self._log_message_prepare(*args)
+        pc_logging.debug(f"Test: {shape.project_name}:{shape.name}: {self.name}{message}")
```

Ruff (0.8.2)

Logging statement uses f-string

(G004)